packages/ec-core/src/test_results.rs

```diff
@@ -231,35 +509,313 @@ pub struct TestResults<R> {
 static_assertions::assert_impl_all!(TestResults<()>: Clone);
 
 impl<R> TestResults<R> {
-    /// Get the number of test results
+    /// Get the number of test results stored in this collection
+    ///
+    /// # Example
+    /// ```
+    /// # use ec_core::test_results::{TestResults, Score};
+    /// let results = [Score(10), Score(20)]
+    ///     .into_iter()
+    ///     .collect::<TestResults<Score<i32>>>();
+    ///
+    /// assert_eq!(results.len(), 2);
+    /// ```
+    #[must_use]
     pub fn len(&self) -> usize {
         self.results.len()
     }
 
-    /// Check if no test results were stored
+    /// Check if this collection is empty
+    ///
+    /// # Examples
+    /// ```
+    /// # use ec_core::test_results::{TestResults, Score};
+    /// let results = [0i32; 0].into_iter().collect::<TestResults<Score<i32>>>();
+    ///
+    /// assert!(results.is_empty());
+    /// ```
+    /// ```
+    /// # use ec_core::test_results::{TestResults, Score};
+    /// let results = [Score(10), Score(20)]
+    ///     .into_iter()
+    ///     .collect::<TestResults<Score<i32>>>();
+    ///
+    /// assert!(!results.is_empty());
+    /// ```
+    #[must_use]
     pub fn is_empty(&self) -> bool {
         self.results.is_empty()
     }
+
+    /// Get a reference to the total result calculated from all the test results
+    /// in this collection.
+    ///
+    /// # Example
+    /// ```
+    /// # use ec_core::test_results::{TestResults, Score};
+    /// let results = [Score(10), Score(20)]
+    ///     .into_iter()
+    ///     .collect::<TestResults<Score<i32>>>();
+    ///
+    /// assert_eq!(results.total(), &Score(30));
+    /// ```
+    #[must_use]
+    pub const fn total(&self) -> &R {
+        &self.total_result
+    }
+
+    /// Get the total result calculated from all the test results
+    /// in this collection, discarding the individual test results
+    ///
+    /// # Example
+    /// ```
+    /// # use ec_core::test_results::{TestResults, Score};
+    /// let results = [Score(10), Score(20)]
+    ///     .into_iter()
+    ///     .collect::<TestResults<Score<i32>>>();
+    ///
+    /// assert_eq!(results.into_total(), Score(30));
+    /// ```
+    #[must_use]
+    pub fn into_total(self) -> R {
```

Comment:
Parts of this PR arguably got a little out of control. These additions to `TestResults` make a lot of sense, but it really would have made more sense if they were in a separate, more focused PR.

Reply:
Fair enough - I think in the process of adding examples and docs I found a few places where apis were like partially implemented (like having a .total() but not a .into_total()) and just went ahead and fully implemented them. I tried not to add too much fully new functionality tho ;)